# PROBLEM 1 : Write a program which will find all such numbers which are symetric (such as 11, 121, 1221,...) between 10 and 2000 (both include). The numbers obtained should be printed in a comma-separated on a single line.

In [64]:
symmetric = [] #list blank 
for i in range(10,2001):
    if i == int(str(i)[::-1]):
        symmetric.append(i)
print(symmetric)

[11, 22, 33, 44, 55, 66, 77, 88, 99, 101, 111, 121, 131, 141, 151, 161, 171, 181, 191, 202, 212, 222, 232, 242, 252, 262, 272, 282, 292, 303, 313, 323, 333, 343, 353, 363, 373, 383, 393, 404, 414, 424, 434, 444, 454, 464, 474, 484, 494, 505, 515, 525, 535, 545, 555, 565, 575, 585, 595, 606, 616, 626, 636, 646, 656, 666, 676, 686, 696, 707, 717, 727, 737, 747, 757, 767, 777, 787, 797, 808, 818, 828, 838, 848, 858, 868, 878, 888, 898, 909, 919, 929, 939, 949, 959, 969, 979, 989, 999, 1001, 1111, 1221, 1331, 1441, 1551, 1661, 1771, 1881, 1991]


# PROBLEM 2 : Write a program which will find all such numbers which include at least 3 duplicated digits (such as 111, 1222,...) between 100 and 2000 (both include). The numbers obtained should be printed in a comma-separated on a single line. 

In [3]:
import collections
duplicate = []
for i in range(99,2001):
    a = str(i)
    for j,count in collections.Counter(a).items():
        if count==3:
            duplicate.append(i)
print(duplicate)

[111, 222, 333, 444, 555, 666, 777, 888, 999, 1000, 1011, 1101, 1110, 1112, 1113, 1114, 1115, 1116, 1117, 1118, 1119, 1121, 1131, 1141, 1151, 1161, 1171, 1181, 1191, 1211, 1222, 1311, 1333, 1411, 1444, 1511, 1555, 1611, 1666, 1711, 1777, 1811, 1888, 1911, 1999, 2000]


# PROBLEM 3 : Write a program will receive an integer larger than 0 and respond number of digits of that integer.

In [4]:
number = int(input("Enter a number larger than 0 : "))
print(len(str(number)))

Enter a number larger than 0 : 12
2


In [5]:
number = int(input("Enter a number larger than 0 : "))
print(len(str(number)))

Enter a number larger than 0 : 021000
5


# PROBLEM 4 : Write a program which will receive an integer larger than 0 and present that integer in binary format.

In [65]:
n=int(input('please enter number : '))
x=n
k=[]
while (n>0):
    a=int(float(n%2))
    k.append(a)
    n=(n-a)/2
k.append(0)
string=""
for j in k[::-1]:
    string=string+str(j)
print('The binary for %d is %s'%(x, string))

please enter number : 1456
The binary for 1456 is 010110110000


In [66]:
n=int(input('please enter number: '))
x=n
k=[]
while (n>0):
    a=int(float(n%2))
    k.append(a)
    n=(n-a)/2
k.append(0)
string=""
for j in k[::-1]:
    string=string+str(j)
print('The binary for %d is %s'%(x, string))

please enter number: 135
The binary for 135 is 010000111


# PROBLEM 5 : Write a program which will receive an integer larger than 100 and find all divisors of that integer. The numbers obtained should be printed in a comma-separated on a single line. 

In [68]:
number = int(input("Enter a number larger than 100: "))
a = [] 
for i in range(1,number+1):
    if number % i == 0:
        a.append(i)    #append: add phần tử vào list
print(a)

Enter a number larger than 100: 269
[1, 269]


In [4]:
number = int(input("Enter a number larger than 100: "))
a = [] 
for i in range(1,number+1):
    if number % i == 0:
        a.append(i)    #append: add phần tử vào list
print(a)

Enter a number larger than 100: 2012
[1, 2, 4, 503, 1006, 2012]


# PROBLEM 6 : Write a program which will find all primes between 1000 and 3000 (both included). The numbers obtained should be printed in a comma-separated on a single line. 

In [5]:
prime = []
for i in range(999,3001):
    count=0
    for j in range(1,i+1):
        if(i%j==0):
            count=count+1
    if(count==2):
        prime.append(i)
print(prime)

[1009, 1013, 1019, 1021, 1031, 1033, 1039, 1049, 1051, 1061, 1063, 1069, 1087, 1091, 1093, 1097, 1103, 1109, 1117, 1123, 1129, 1151, 1153, 1163, 1171, 1181, 1187, 1193, 1201, 1213, 1217, 1223, 1229, 1231, 1237, 1249, 1259, 1277, 1279, 1283, 1289, 1291, 1297, 1301, 1303, 1307, 1319, 1321, 1327, 1361, 1367, 1373, 1381, 1399, 1409, 1423, 1427, 1429, 1433, 1439, 1447, 1451, 1453, 1459, 1471, 1481, 1483, 1487, 1489, 1493, 1499, 1511, 1523, 1531, 1543, 1549, 1553, 1559, 1567, 1571, 1579, 1583, 1597, 1601, 1607, 1609, 1613, 1619, 1621, 1627, 1637, 1657, 1663, 1667, 1669, 1693, 1697, 1699, 1709, 1721, 1723, 1733, 1741, 1747, 1753, 1759, 1777, 1783, 1787, 1789, 1801, 1811, 1823, 1831, 1847, 1861, 1867, 1871, 1873, 1877, 1879, 1889, 1901, 1907, 1913, 1931, 1933, 1949, 1951, 1973, 1979, 1987, 1993, 1997, 1999, 2003, 2011, 2017, 2027, 2029, 2039, 2053, 2063, 2069, 2081, 2083, 2087, 2089, 2099, 2111, 2113, 2129, 2131, 2137, 2141, 2143, 2153, 2161, 2179, 2203, 2207, 2213, 2221, 2237, 2239, 2243, 225

# PROBLEM 7 : Write a program which will receive an integer larger than 0 and respond all perfect numbers smaller than that integer. A perfect number is sum of its divisors except itself, such as 6 = 1+2+3. The numbers obtained should be printed in a comma-separated sequence on a single line.

In [40]:
d=[]
def perfect(n):
    
    if n<1:
        return False

    perfect_sum = 0
    
    for i in range(1,n):
        if n%i==0:
            perfect_sum += i
            
    return perfect_sum == n

# Reading interval from user
value = int(input('Enter value: '))

# Looping & displaying if it is Perfect
print('Perfect numbers smaller than %d are:' %(value) )
for i in range(0,value+1):
    if perfect(i):
        d.append(i)
print(d, end=' ')

Enter value: 1000
Perfect numbers smaller than 1000 are:
[6, 28, 496] 

In [38]:
d=[]
def perfect(n):
    
    if n<1:
        return False

    perfect_sum = 0
    
    for i in range(1,n):
        if n%i==0:
            perfect_sum += i
            
    return perfect_sum == n

# Reading interval from user
value = int(input('Enter value: '))

# Looping & displaying if it is Perfect
print('Perfect numbers smaller than %d are:' %(value) )
for i in range(0,value+1):
    if perfect(i):
        d.append(i)
print(d, end=' ')

Enter value: 100
Perfect numbers smaller than 100 are:
[6, 28] 

# PROBLEM 8 : With a given integral number n, write a program to generate a dictionary that contains (i, iii) such that is an integral number between 1 and n (both included). and then the program should print the dictionary. Suppose the following input is supplied to the program: 6 Then, the output should be: {1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216}

In [47]:
    dict={}
    n = int(input("Enter a number: "))
    for i in range(1,n+1):
        dict[i]=i*i*i
print (dict)

Enter a number: 7
{1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216, 7: 343}


# PROBLEM 9 : Write a program which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 3 or not. The numbers that are divisible by 3 are to be printed in a comma separated sequence. Example: 0100,0011,1010,1101 Then the output should be: 1101

In [33]:
a = []
num = [x for x in input().split(',')]
for p in num:
    x = int(p)
    if not x%3:
        a.append(p)
print(','.join(a))

0100,0011,1010,1101
1101


# PROBLEM 10 :Write a program which will receive an integer larger than 0 and respond if that integer is sum of three distinct squared numbers, such as 38 = 2^2 + 3^2 + 5^2

In [1]:
def sumSquare( n) : 
    i = 1 
    while i * i <= n : 
        j = 1
        while ((j * j <= n) and (j != i)): 
            k = 1
            while((k * k <= n) and (k != i)and(j != k) ) :
                if (i * i + j * j + k * k == n) : 
                    print(n,"=",i, "^2 + ", j , "^2+ ", k , "^2" ) 
                    return True
                k = k + 1
            j = j + 1
        i = i + 1
          
    return False
   
# driver Program 
n = int(input("number "))
if (sumSquare(n)) : 
    print("Yes") 
else : 
    print( "No")

number 38
38 = 5 ^2 +  3 ^2+  2 ^2
Yes


In [2]:
def sumSquare( n) : 
    i = 1 
    while i * i <= n : 
        j = 1
        while ((j * j <= n) and (j != i)): 
            k = 1
            while((k * k <= n) and (k != i)and(j != k) ) :
                if (i * i + j * j + k * k == n) : 
                    print(n,"=",i, "^2 + ", j , "^2+ ", k , "^2" ) 
                    return True
                k = k + 1
            j = j + 1
        i = i + 1
          
    return False
   
# driver Program 
n = int(input("number "))
if (sumSquare(n)) : 
    print("Yes") 
else : 
    print( "No")


number 10
No


In [3]:
def sumSquare( n) : 
    i = 1 
    while i * i <= n : 
        j = 1
        while ((j * j <= n) and (j != i)): 
            k = 1
            while((k * k <= n) and (k != i)and(j != k) ) :
                if (i * i + j * j + k * k == n) : 
                    print(n,"=",i, "^2 + ", j , "^2+ ", k , "^2" ) 
                    return True
                k = k + 1
            j = j + 1
        i = i + 1
          
    return False
   
# driver Program 
n = int(input("number "))
if (sumSquare(n)) : 
    print("Yes") 
else : 
    print( "No")


number 121
121 = 9 ^2 +  6 ^2+  2 ^2
Yes
